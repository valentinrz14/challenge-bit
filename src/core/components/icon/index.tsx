import React, { FunctionComponent } from 'react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

/*
 ** Types
 */

interface ChallengeBitIconProps {
  iconProps: {
    name: string;
    size: number;
    color: string;
  };
}

/*
 ** Component
 */

export const ChallengeBitIcon: FunctionComponent<ChallengeBitIconProps> = ({
  iconProps: { name, size, color },
}) => {
  return <Icon name={name} size={size} color={color} />;
};
