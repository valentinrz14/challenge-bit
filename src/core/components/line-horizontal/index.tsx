import React, { FunctionComponent } from 'react';
import { StyleSheet, View } from 'react-native';

/*
 ** Types
 */

interface LineHorizontalProps {
  height: number;
  bgColor: string;
}
/*
 ** Styles
 */
const styles = StyleSheet.create({
  line: {
    width: '100%',
  },
});
/*
 ** Component
 */

export const LineHorizontal: FunctionComponent<LineHorizontalProps> = ({
  bgColor,
  height,
}) => {
  return <View style={[styles.line, { backgroundColor: bgColor, height }]} />;
};
