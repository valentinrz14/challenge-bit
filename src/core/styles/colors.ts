export const Colors = {
  black: '#000000',
  white: '#FFFFFF',
  greyLight: '#F6F6F6',
  greyMedium: '#C5C6C8',
  grey: '#A3A5A8',
  greyDark: '#7D7E81',
  green: '#00AA00',
  red: '#FF0000',
  blue: '#337BF6',
};
