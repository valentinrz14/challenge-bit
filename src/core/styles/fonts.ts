export const Fonts = {
  HubballiRegular: 'Hubballi-Regular',
  MontserratRegular: 'Montserrat-Regular',
  MontserratMedium: 'Montserrat-Medium',
  MontserratBold: 'Montserrat-Bold',
  MontserratLight: 'Montserrat-Light',
  MontserratSemiBold: 'Montserrat-SemiBold',
};

export type FontsTypes = typeof Fonts;
