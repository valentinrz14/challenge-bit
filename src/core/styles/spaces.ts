export const MarginsHorizontal = {
  small: 5,
  medium: 10,
  large: 20,
  extraLarge: 30,
};
export const MarginsVertical = {
  small: 5,
  medium: 10,
  large: 15,
  extraLarge: 20,
};

export const PaddingsVertical = {
  small: 5,
  medium: 10,
  large: 15,
  extraLarge: 20,
};

export const PaddingsHorizontal = {
  small: 5,
  medium: 10,
  large: 15,
  extraLarge: 20,
};
