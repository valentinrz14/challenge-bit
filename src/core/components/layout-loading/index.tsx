import { Colors } from '@core/styles/colors';
import React, { FunctionComponent } from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';

/*
 ** Types
 */
interface LayoutLoadingProps {
  color?: string;
  size?: 'large' | 'small';
}
/*
 ** Styles
 */

const styles = StyleSheet.create({
  wrapperLayoutLoading: {
    marginTop: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
/*
 ** Component
 */
export const LayoutLoading: FunctionComponent<LayoutLoadingProps> = ({
  color = Colors.blue,
  size = 'large',
}) => {
  return (
    <View style={styles.wrapperLayoutLoading}>
      <ActivityIndicator size={size} color={color} />
    </View>
  );
};
