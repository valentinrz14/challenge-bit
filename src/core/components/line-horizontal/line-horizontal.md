```TSX
import React, { FunctionComponent } from 'react';
import { ChallengeBitIcon } from './index.tsx';
import { Colors } from '@core/styles/colors';

export const App: FunctionComponent = () => ( 
    <LineHorizontal height={1} bgColor={Colors.black} />
)
```
