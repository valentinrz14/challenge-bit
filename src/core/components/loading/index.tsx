import { Colors } from '@core/styles/colors';
import React, { FunctionComponent } from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';

/*
 ** Types
 */
interface LoadingProps {
  color?: string;
  size?: 'large' | 'small';
}
/*
 ** Styles
 */

const styles = StyleSheet.create({
  wrapperLayoutLoading: {
    marginTop: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
/*
 ** Component
 */
export const Loading: FunctionComponent<LoadingProps> = ({
  color = Colors.blue,
  size = 'large',
}) => {
  return (
    <View style={styles.wrapperLayoutLoading}>
      <ActivityIndicator size={size} color={color} />
    </View>
  );
};
