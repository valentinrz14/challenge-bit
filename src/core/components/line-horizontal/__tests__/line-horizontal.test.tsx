import { LineHorizontal } from '@core/components/line-horizontal';
import { Colors } from '@core/styles/colors';
import { render } from '@testing-library/react-native';
import React from 'react';

describe('LineHorizontal', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <LineHorizontal height={1} bgColor={Colors.black} />,
    );
    expect(toJSON).toMatchSnapshot();
  });
});
