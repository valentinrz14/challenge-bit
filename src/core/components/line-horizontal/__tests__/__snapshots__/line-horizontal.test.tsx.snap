// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineHorizontal renders correctly 1`] = `[Function]`;
